# Importing Necessary Library in Python

In [22]:
# Importing necesary Library in Python 

import numpy as np
import pandas as pd 
import matplotlib.pyplot as plt
import matplotlib.dates as mdates
%matplotlib inline 
import seaborn as sns
import datetime
import scipy.stats as stats
# Standard libs
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline
from matplotlib.gridspec import GridSpec
pd.set_option('display.max_columns', 100)
import plotly.offline as py
import plotly.express as px
import plotly.graph_objs as go
import json
import requests
import folium
#import opendatasets as od
from datetime import datetime
import calendar
from pandas.api.types import CategoricalDtype
import warnings
warnings.filterwarnings("ignore")

#from wordcloud import WordCloud
from collections import Counter
from PIL import Image

# Data Reading 

In [2]:
# Reading Brazilian Ecommerce Datasets in Python
import time
start = time.time()

data = pd.read_csv("E:/IIMN/MODULE 02/ASSIGNMENT/archive (8)/olist_customers_dataset.csv")
geo_data = pd.read_csv("E:/IIMN/MODULE 02/ASSIGNMENT/archive (8)/olist_geolocation_dataset.csv")
order_itemdata = pd.read_csv("E:/IIMN/MODULE 02/ASSIGNMENT/archive (8)/olist_order_items_dataset.csv")
pay_data = pd.read_csv("E:/IIMN/MODULE 02/ASSIGNMENT/archive (8)/olist_order_payments_dataset.csv")
rev_data = pd.read_csv("E:/IIMN/MODULE 02/ASSIGNMENT/archive (8)/olist_order_reviews_dataset.csv")
orders = pd.read_csv("E:/IIMN/MODULE 02/ASSIGNMENT/archive (8)/olist_orders_dataset.csv")
order_selldata=pd.read_csv("E:/IIMN/MODULE 02/ASSIGNMENT/archive (8)/olist_sellers_dataset.csv")
order_prddata=pd.read_csv("E:/IIMN/MODULE 02/ASSIGNMENT/archive (8)/olist_products_dataset.csv")
order_prd_catdata=pd.read_csv("E:/IIMN/MODULE 02/ASSIGNMENT/archive (8)/product_category_name_translation.csv")

end = time.time()
print("reading time: ",(end-start),"sec")

reading time:  2.2513904571533203 sec


# Data Prepossing 

In [3]:
#checking number of columns , column_names and no_of_rows

datasets = [data,geo_data,order_itemdata, pay_data, orders, order_prddata,rev_data,order_selldata, order_prd_catdata]
titles = ["customers","geolocations","items", "payments", "orders", "products","reviews","sellers","category_translation"]



info_df = pd.DataFrame({},)
info_df['dataset']= titles

info_df['no_of_columns']= [len(df.columns) for df in datasets ]
info_df['columns_name']= [', '.join(list(df.columns)) for df in datasets] 
info_df['no_of_rows'] = [len(df) for df in datasets]

info_df.style.background_gradient(cmap='Greys')

### Observations 

Dataset with Maximum No fo Columns is Products
Dataset with Maximum No of Rows is Geolocations

In [4]:
#checking dtypes
datasets = [data,geo_data,order_itemdata, pay_data, orders, order_prddata,rev_data,order_selldata, order_prd_catdata]
titles = ["customers","geolocations","items", "payments", "orders", "products","reviews","sellers","category_translation"]

numerics = ['int16', 'int32', 'int64', 'float16', 'float32', 'float64']
new_df = pd.DataFrame({},)
new_df['dataset']= titles

new_df['numeric_features'] = [len((df.select_dtypes(include=numerics)).columns) for df in datasets]
new_df['num_features_name'] = [', '.join(list((df.select_dtypes(include=numerics)).columns)) for df in datasets]
new_df['object_features'] = [len((df.select_dtypes(include='object')).columns) for df in datasets]
new_df['objt_features_name'] = [', '.join(list((df.select_dtypes(include='object')).columns)) for df in datasets]
new_df['bool_features'] = [len((df.select_dtypes(include='bool')).columns) for df in datasets]
new_df.style.background_gradient(cmap='Greys')


### Observation(s):

products dataset has maximum number of numerical features(i.e dtype :'int16', 'int32', 'int64', 'float16', 'float32', 'float64').

orders dataset has maximum number of features of object dtype

We can also observe that all the timestamps are in object datatypes.So, we have to convert it into datetime type to do analysis on these features.

###### Checking Null Values in Datasets

In [6]:
#checking no of null values
#code source-https://www.kaggle.com/fayhosseini/brazilian-e-commerce-eda-for-beginners

datasets = [data,geo_data,order_itemdata, pay_data, orders, order_prddata,rev_data,order_selldata, order_prd_catdata]
titles = ["customers","geolocations","items", "payments", "orders", "products","reviews","sellers","category_translation"]

info_df_n = pd.DataFrame({},)

info_df_n['dataset']= titles

#creating column of column names in the dataset 
info_df_n['cols'] = [', '.join([col for col, null in df.isnull().sum().items() ]) for df in datasets]

#creating total number of columns in the dataset 
info_df_n['cols_no']= [df.shape[1] for df in datasets]

#counting total null values
info_df_n['null_no']= [df.isnull().sum().sum() for df in datasets]

#creating total number of columns in the dataset with null-values 
info_df_n['null_cols_no']= [len([col for col, null in df.isnull().sum().items() if null > 0]) for df in datasets]

#creating column of name of columns in the dataset with null-values 
info_df_n['null_cols'] = [', '.join([col for col, null in df.isnull().sum().items() if null > 0]) for df in datasets]


info_df_n.style.background_gradient(cmap='Greys')

### Observation(s):

The maximum number of null-values are present in reviews dataset and the name of the columns with the null-values are review_comment_title and review_comment_message.

products dataset contains least number of null- values but most of its columns has null-values.
we have to deal with these null-values in future.

#### Merging Datasets

In [5]:
rev_new = rev_data.drop(['review_comment_title','review_creation_date','review_id','review_answer_timestamp'],axis=1)

In [6]:
df = pd.merge(orders,pay_data, on="order_id")
df = df.merge(data, on="customer_id")
df = df.merge(order_itemdata, on="order_id")
df = df.merge(order_prddata, on="product_id")
df = df.merge(order_prd_catdata, on="product_category_name")
df = df.merge(rev_new, on="order_id")
df.head()


order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
1  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
2  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
3  128e10d95713541c87cd1a2e48201934  a20e8105f23924cd00833fd87daa0831   
4  0e7e841ddf8f8f2de2bad69267ecfbcf  26c7ac168e1433912a51b924fbd34d34   

  order_status order_purchase_timestamp    order_approved_at  \
0    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
1    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
2    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
3    delivered      2017-08-15 18:29:31  2017-08-15 20:05:16   
4    delivered      2017-08-02 18:24:47  2017-08-02 18:43:15   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2017-10-10 21:25:13   
1          2017-10-04 19:55:00           2017-10-10 21:25:13   
2          2017-10-04 19:55:00           2017-10-10 21:25:13   
3          2017-08-17 15:28:33           2017-08-18 14:44:43   
4          2017-08-04 17:35:43           2017-08-07 18:30:01   

  order_estimated_delivery_date  payment_sequential payment_type  \
0           2017-10-18 00:00:00                   1  credit_card   
1           2017-10-18 00:00:00                   3      voucher   
2           2017-10-18 00:00:00                   2      voucher   
3           2017-08-28 00:00:00                   1  credit_card   
4           2017-08-15 00:00:00                   1  credit_card   

   payment_installments  payment_value                customer_unique_id  \
0                     1          18.12  7c396fd4830fd04220f754e42b4e5bff   
1                     1           2.00  7c396fd4830fd04220f754e42b4e5bff   
2                     1          18.59  7c396fd4830fd04220f754e42b4e5bff   
3                     3          37.77  3a51803cc0d012c3b5dc8b7528cb05f7   
4                     1          37.77  ef0996a1a279c26e7ecbd737be23d235   

   customer_zip_code_prefix customer_city customer_state  order_item_id  \
0                      3149     sao paulo             SP              1   
1                      3149     sao paulo             SP              1   
2                      3149     sao paulo             SP              1   
3                      3366     sao paulo             SP              1   
4                      2290     sao paulo             SP              1   

                         product_id                         seller_id  \
0  87285b34884572647811a353c7ac498a  3504c0cb71d7fa48d967e0e4c94d59d9   
1  87285b34884572647811a353c7ac498a  3504c0cb71d7fa48d967e0e4c94d59d9   
2  87285b34884572647811a353c7ac498a  3504c0cb71d7fa48d967e0e4c94d59d9   
3  87285b34884572647811a353c7ac498a  3504c0cb71d7fa48d967e0e4c94d59d9   
4  87285b34884572647811a353c7ac498a  3504c0cb71d7fa48d967e0e4c94d59d9   

   shipping_limit_date  price  freight_value  product_category_name  \
0  2017-10-06 11:07:15  29.99           8.72  utilidades_domesticas   
1  2017-10-06 11:07:15  29.99           8.72  utilidades_domesticas   
2  2017-10-06 11:07:15  29.99           8.72  utilidades_domesticas   
3  2017-08-21 20:05:16  29.99           7.78  utilidades_domesticas   
4  2017-08-08 18:37:31  29.99           7.78  utilidades_domesticas   

   product_name_lenght  product_description_lenght  product_photos_qty  \
0                 40.0                       268.0                 4.0   
1                 40.0                       268.0                 4.0   
2                 40.0                       268.0                 4.0   
3                 40.0                       268.0                 4.0   
4                 40.0                       268.0                 4.0   

   product_weight_g  product_length_cm  product_height_cm  product_width_cm  \
0             500.0               19.0                8.0              13.0   
1             500.0             

In [7]:
print("Number of rows after merging:",len(df))
print("Number of columns after merging:",len(df.columns))

Number of rows after merging: 115609
Number of columns after merging: 33


#### Taking Care of Missing values 

In [8]:
df.isnull().sum()

order_id                             0
customer_id                          0
order_status                         0
order_purchase_timestamp             0
order_approved_at                   14
order_delivered_carrier_date      1195
order_delivered_customer_date     2400
order_estimated_delivery_date        0
payment_sequential                   0
payment_type                         0
payment_installments                 0
payment_value                        0
customer_unique_id                   0
customer_zip_code_prefix             0
customer_city                        0
customer_state                       0
order_item_id                        0
product_id                           0
seller_id                            0
shipping_limit_date                  0
price                                0
freight_value                        0
product_category_name                0
product_name_lenght                  0
product_description_lenght           0
product_photos_qty       

###### Handling Missing values in Timestamps

The order of different types of timestamps are shown below:
order_purchase_timestamp-->order_approved_at--> order_delivered_carrier_date-->order_delivered_customer_date-->order_estimated_delivery_date     

Timestamps containg missing values are order_approved_at, order_delivered_carrier_date, order_delivered_customer_date.

null-values in order_approved_at can be replaced by order_purchase_timestamp and null-values in order_delivered_customer_date can be replaced by order_estimated_delivery_date

we can drop the column order_delivered_carrier_date.

In [9]:
#Handling missing values
index = (df[df['order_delivered_customer_date'].isnull() == True].index.values)

df["order_approved_at"].fillna(df["order_purchase_timestamp"], inplace=True)
df["order_delivered_customer_date"].fillna(df["order_estimated_delivery_date"], inplace=True)

#dropping order delivery carrier date
df.drop(labels='order_delivered_carrier_date',axis=1,inplace=True)

In [12]:
#checking the replaced values
df.order_estimated_delivery_date[index[0]]

'2017-07-06 00:00:00'

In [13]:
df.order_delivered_customer_date[index[0]]

'2017-07-06 00:00:00'

In [10]:
# Handling missing values of numerical features
df['product_weight_g'].fillna(df['product_weight_g'].median(),inplace=True)
df['product_length_cm'].fillna(df['product_length_cm'].median(),inplace=True)
df['product_height_cm'].fillna(df['product_height_cm'].median(),inplace=True)
df['product_width_cm'].fillna(df['product_width_cm'].median(),inplace=True)

In [11]:
#Handling missing values of text column
print("Percentage of null reviews :",(df.review_comment_message.isnull().sum()/len(df))*100 ,"%")
# filling null value of review comments with no_review
df['review_comment_message'].fillna('nao_reveja',inplace=True)

Percentage of null reviews : 57.697065107387836 %


###### 3.1.4 Data Dedublicate

In [12]:
dup_rows = df[df.duplicated(['order_id','customer_id','order_purchase_timestamp','order_delivered_customer_date','customer_unique_id','review_comment_message'])]
dup_rows.head()

order_id                       customer_id  \
1   e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
2   e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
8   82bce245b1c9148f8d19a55b9ff70644  388025bec8128ff20ec1a316ed4dcf02   
9   82bce245b1c9148f8d19a55b9ff70644  388025bec8128ff20ec1a316ed4dcf02   
10  82bce245b1c9148f8d19a55b9ff70644  388025bec8128ff20ec1a316ed4dcf02   

   order_status order_purchase_timestamp    order_approved_at  \
1     delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
2     delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
8     delivered      2017-04-20 17:15:46  2017-04-21 05:15:56   
9     delivered      2017-04-20 17:15:46  2017-04-21 05:15:56   
10    delivered      2017-04-20 17:15:46  2017-04-21 05:15:56   

   order_delivered_customer_date order_estimated_delivery_date  \
1            2017-10-10 21:25:13           2017-10-18 00:00:00   
2            2017-10-10 21:25:13           2017-10-18 00:00:00   
8            2017-05-10 09:17:55           2017-05-12 00:00:00   
9            2017-05-10 09:17:55           2017-05-12 00:00:00   
10           2017-05-10 09:17:55           2017-05-12 00:00:00   

    payment_sequential payment_type  payment_installments  payment_value  \
1                    3      voucher                     1           2.00   
2                    2      voucher                     1          18.59   
8                    1       boleto                     1         267.80   
9                    1       boleto                     1         267.80   
10                   1       boleto                     1         267.80   

                  customer_unique_id  customer_zip_code_prefix customer_city  \
1   7c396fd4830fd04220f754e42b4e5bff                      3149     sao paulo   
2   7c396fd4830fd04220f754e42b4e5bff                      3149     sao paulo   
8   f9effeed3df9ae063a58c0759b96f8b2                     85804      cascavel   
9   f9effeed3df9ae063a58c0759b96f8b2                     85804      cascavel   
10  f9effeed3df9ae063a58c0759b96f8b2                     85804      cascavel   

   customer_state  order_item_id                        product_id  \
1              SP              1  87285b34884572647811a353c7ac498a   
2              SP              1  87285b34884572647811a353c7ac498a   
8              PR              2  a5a0e71a81ae65aa335e71c06261e260   
9              PR              3  a5a0e71a81ae65aa335e71c06261e260   
10             PR              4  a5a0e71a81ae65aa335e71c06261e260   

                           seller_id  shipping_limit_date  price  \
1   3504c0cb71d7fa48d967e0e4c94d59d9  2017-10-06 11:07:15  29.99   
2   3504c0cb71d7fa48d967e0e4c94d59d9  2017-10-06 11:07:15  29.99   
8   c8417879a15366a17c30af34c798c332  2017-04-27 05:15:56  38.00   
9   c8417879a15366a17c30af34c798c332  2017-04-27 05:15:56  38.00   
10  c8417879a15366a17c30af34c798c332  2017-04-27 05:15:56  38.00   

    freight_value  product_category_name  product_name_lenght  \
1            8.72  utilidades_domesticas                 40.0   
2            8.72  utilidades_domesticas                 40.0   
8           15.56  utilidades_domesticas                 57.0   
9           15.56  utilidades_domesticas                 57.0   
10          15.56  utilidades_domesticas                 57.0   

    product_description_lenght  product_photos_qty  product_weight_g  \
1                        268.0                 4.0             500.0   
2                        268.0                 4.0             500.0   
8                        698.0                 3.0             705.0   
9                        698.0                 3.0             705.0   
10                       698.0                 3.0             705.0   

    product_length_cm  product_height_cm  product_width_cm  \
1                19.0                8.0              13.0   
2                19.0                8.0              13.0   
8                

In [13]:
#Deduplication of entries
df= df.drop_duplicates(subset={'order_id','customer_id','order_purchase_timestamp','order_delivered_customer_date'}, keep='first', inplace=False)
df=df.reindex()
df

order_id                       customer_id  \
0       e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
3       128e10d95713541c87cd1a2e48201934  a20e8105f23924cd00833fd87daa0831   
4       0e7e841ddf8f8f2de2bad69267ecfbcf  26c7ac168e1433912a51b924fbd34d34   
5       bfc39df4f36c3693ff3b63fcbea9e90a  53904ddbea91e1e92b2b3f1d09a7af86   
6       6ea2f835b4556291ffdc53fa0b3b95e8  c7340080e394356141681bd4c9b8fe31   
...                                  ...                               ...   
115603  2cfc3681e512e0d12b6fe84b396e3907  6c60b00bcbef919c1697d18c309a99fe   
115604  0b82d0616f1ad8da15cf967b984b4004  986632b40c38f4240caf8608cb01d40d   
115605  2ef4a11b6e24fdfbb43b92cb5f95edff  ee1cfdc92e449920e25d3ca4ab4da4f6   
115607  2c4ada2e75c2ad41dd93cebb5df5f023  363d3a9b2ec5c5426608688ca033292d   
115608  bede3503afed051733eeb4a84d1adcc5  919570a26efbd068d6a0f66d5c5072a3   

       order_status order_purchase_timestamp    order_approved_at  \
0         delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
3         delivered      2017-08-15 18:29:31  2017-08-15 20:05:16   
4         delivered      2017-08-02 18:24:47  2017-08-02 18:43:15   
5         delivered      2017-10-23 23:26:46  2017-10-25 02:14:11   
6         delivered      2017-11-24 21:27:48  2017-11-25 00:21:09   
...             ...                      ...                  ...   
115603    delivered      2018-04-22 20:15:52  2018-04-24 18:46:28   
115604    delivered      2018-08-03 21:35:53  2018-08-04 04:25:14   
115605    delivered      2018-07-23 18:35:14  2018-07-24 10:31:25   
115607    delivered      2017-01-26 11:09:00  2017-01-26 11:22:17   
115608    delivered      2017-09-17 16:51:43  2017-09-19 04:05:52   

       order_delivered_customer_date order_estimated_delivery_date  \
0                2017-10-10 21:25:13           2017-10-18 00:00:00   
3                2017-08-18 14:44:43           2017-08-28 00:00:00   
4                2017-08-07 18:30:01           2017-08-15 00:00:00   
5                2017-11-07 18:04:59           2017-11-13 00:00:00   
6                2017-12-28 18:59:23           2017-12-21 00:00:00   
...                              ...                           ...   
115603           2018-04-30 20:51:45           2018-05-18 00:00:00   
115604           2018-08-09 21:42:08           2018-08-10 00:00:00   
115605           2018-07-27 14:04:10           2018-08-06 00:00:00   
115607           2017-02-14 16:24:01           2017-03-07 00:00:00   
115608           2017-09-28 18:14:35           2017-10-11 00:00:00   

        payment_sequential payment_type  payment_installments  payment_value  \
0                        1  credit_card                     1          18.12   
3                        1  credit_card                     3          37.77   
4                        1  credit_card                     1          37.77   
5                        1       boleto                     1          44.09   
6                        1  credit_card                    10         356.12   
...                    ...          ...                   ...            ...   
115603                   1  credit_card                     1          34.22   
115604                   1       boleto                     1          33.69   
115605                   1  credit_card                     1          84.63   
115607                   1  credit_card                     1         209.06   
115608                   1       boleto                     1         115.45   

                      customer_unique_id  customer_zip_code_prefix  \
0       7c396fd4830fd04220f754e42b4e5bff                      3149   
3       3a51803cc0d012c3b5dc8b7528cb05f7                      3366   
4       ef0996a1a279c26e7ecbd737be23d235                      2290   
5       e781fdcc107d13d865fc7698711cc572                     88032   
6       3e4fd73f1e86b135b9b121d6abbe9597                     19400   
...                                  ...    

In [14]:
print("Number of rows after dedublication:",len(df))
print("Number of columns after deduplication:",len(df.columns))

Number of rows after dedublication: 96516
Number of columns after deduplication: 32


# Data Analysis

In [15]:
# all time stamps are in object dtype as observed above converting it into dataetime 
df[['order_purchase_timestamp','order_approved_at','order_delivered_customer_date','order_estimated_delivery_date',]]=df[['order_purchase_timestamp',
       'order_approved_at','order_delivered_customer_date','order_estimated_delivery_date']].apply(pd.to_datetime)



In [16]:
df.isnull().sum()

order_id                         0
customer_id                      0
order_status                     0
order_purchase_timestamp         0
order_approved_at                0
order_delivered_customer_date    0
order_estimated_delivery_date    0
payment_sequential               0
payment_type                     0
payment_installments             0
payment_value                    0
customer_unique_id               0
customer_zip_code_prefix         0
customer_city                    0
customer_state                   0
order_item_id                    0
product_id                       0
seller_id                        0
shipping_limit_date              0
price                            0
freight_value                    0
product_category_name            0
product_name_lenght              0
product_description_lenght       0
product_photos_qty               0
product_weight_g                 0
product_length_cm                0
product_height_cm                0
product_width_cm    

Observation(s):

the final merged dataset has no null values.
total number of columns is 32.
  dtype           |   number of columns
  ----------------|-------------------
  datetime64[ns]  |      4
                  |
  float64(10)     |      10
                  |
  int64           |      5
                  |
  object          |      13 

In [17]:
df.describe()

payment_sequential  payment_installments  payment_value  \
count        96516.000000          96516.000000   96516.000000   
mean             1.022545              2.919858     157.924421   
std              0.247935              2.711997     216.773702   
min              1.000000              0.000000       0.010000   
25%              1.000000              1.000000      60.065000   
50%              1.000000              2.000000     103.190000   
75%              1.000000              4.000000     175.392500   
max             27.000000             24.000000   13664.080000   

       customer_zip_code_prefix  order_item_id         price  freight_value  \
count              96516.000000   96516.000000  96516.000000   96516.000000   
mean               35163.974543       1.016205    125.777851      20.220468   
std                29810.746531       0.145692    189.535209      15.931328   
min                 1003.000000       1.000000      0.850000       0.000000   
25%                11368.750000       1.000000     41.900000      13.300000   
50%                24422.000000       1.000000     79.000000      16.390000   
75%                59015.000000       1.000000    139.900000      21.260000   
max                99980.000000       7.000000   6735.000000     409.680000   

       product_name_lenght  product_description_lenght  product_photos_qty  \
count         96516.000000                96516.000000        96516.000000   
mean             48.846253                  794.118188            2.251015   
std               9.995674                  654.863686            1.746647   
min               5.000000                    4.000000            1.000000   
25%              42.000000                  349.000000            1.000000   
50%              52.000000                  607.000000            2.000000   
75%              57.000000                  996.000000            3.000000   
max              76.000000                 3992.000000           20.000000   

       product_weight_g  product_length_cm  product_height_cm  \
count      96516.000000       96516.000000       96516.000000   
mean        2106.227133          30.147955          16.507771   
std         3764.648279          16.154906          13.339701   
min            0.000000           7.000000           2.000000   
25%          300.000000          18.000000           8.000000   
50%          700.000000          25.000000          13.000000   
75%         1813.000000          38.000000          20.000000   
max        40425.000000         105.000000         105.000000   

       product_width_cm  review_score  
count      96516.000000  96516.000000  
mean          23.061855      4.107412  
std           11.744019      1.329213  
min            6.000000      1.000000  
25%           15.000000      4.000000  
50%           20.000000      5.000000  
75%           30.000000      5.000000  
max          118.000000      5.000000

###### Observation(s):

We can observe from the above table except customer_zip_code_prefix, order_item_id and review_score features we have 12 numerical features in our final dataset.

Also,We can observe the statistics like percentile values , mean and standard deviation values, count , min and max of the numerical featues.For payment_value, the maximum payment value of an order is 13664 Brazilian real.

For the price and freight value of an order. The maximum price of an order is 6735 while max freight is around 410 Brazilian real. The average price of an order is around 125 Brazilian real and frieght value is around 20 Brazilian real. The order with minimum price of 0.85 Brazilian real have been made.

Similarly, we can observe the other features further we will see the distribution of these features and see how they are helping in classifying the class labels and find other insights.

In [20]:
# checking the target variables i.e review score 
df.review_score.value_counts()

5    65374
4    21951
1    14546
3     9718
2     4020
Name: review_score, dtype: int64

In [18]:
df

order_id                       customer_id  \
0       e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
3       128e10d95713541c87cd1a2e48201934  a20e8105f23924cd00833fd87daa0831   
4       0e7e841ddf8f8f2de2bad69267ecfbcf  26c7ac168e1433912a51b924fbd34d34   
5       bfc39df4f36c3693ff3b63fcbea9e90a  53904ddbea91e1e92b2b3f1d09a7af86   
6       6ea2f835b4556291ffdc53fa0b3b95e8  c7340080e394356141681bd4c9b8fe31   
...                                  ...                               ...   
115603  2cfc3681e512e0d12b6fe84b396e3907  6c60b00bcbef919c1697d18c309a99fe   
115604  0b82d0616f1ad8da15cf967b984b4004  986632b40c38f4240caf8608cb01d40d   
115605  2ef4a11b6e24fdfbb43b92cb5f95edff  ee1cfdc92e449920e25d3ca4ab4da4f6   
115607  2c4ada2e75c2ad41dd93cebb5df5f023  363d3a9b2ec5c5426608688ca033292d   
115608  bede3503afed051733eeb4a84d1adcc5  919570a26efbd068d6a0f66d5c5072a3   

       order_status order_purchase_timestamp   order_approved_at  \
0         delivered      2017-10-02 10:56:33 2017-10-02 11:07:15   
3         delivered      2017-08-15 18:29:31 2017-08-15 20:05:16   
4         delivered      2017-08-02 18:24:47 2017-08-02 18:43:15   
5         delivered      2017-10-23 23:26:46 2017-10-25 02:14:11   
6         delivered      2017-11-24 21:27:48 2017-11-25 00:21:09   
...             ...                      ...                 ...   
115603    delivered      2018-04-22 20:15:52 2018-04-24 18:46:28   
115604    delivered      2018-08-03 21:35:53 2018-08-04 04:25:14   
115605    delivered      2018-07-23 18:35:14 2018-07-24 10:31:25   
115607    delivered      2017-01-26 11:09:00 2017-01-26 11:22:17   
115608    delivered      2017-09-17 16:51:43 2017-09-19 04:05:52   

       order_delivered_customer_date order_estimated_delivery_date  \
0                2017-10-10 21:25:13                    2017-10-18   
3                2017-08-18 14:44:43                    2017-08-28   
4                2017-08-07 18:30:01                    2017-08-15   
5                2017-11-07 18:04:59                    2017-11-13   
6                2017-12-28 18:59:23                    2017-12-21   
...                              ...                           ...   
115603           2018-04-30 20:51:45                    2018-05-18   
115604           2018-08-09 21:42:08                    2018-08-10   
115605           2018-07-27 14:04:10                    2018-08-06   
115607           2017-02-14 16:24:01                    2017-03-07   
115608           2017-09-28 18:14:35                    2017-10-11   

        payment_sequential payment_type  payment_installments  payment_value  \
0                        1  credit_card                     1          18.12   
3                        1  credit_card                     3          37.77   
4                        1  credit_card                     1          37.77   
5                        1       boleto                     1          44.09   
6                        1  credit_card                    10         356.12   
...                    ...          ...                   ...            ...   
115603                   1  credit_card                     1          34.22   
115604                   1       boleto                     1          33.69   
115605                   1  credit_card                     1          84.63   
115607                   1  credit_card                     1         209.06   
115608                   1       boleto                     1         115.45   

                      customer_unique_id  customer_zip_code_prefix  \
0       7c396fd4830fd04220f754e42b4e5bff                      3149   
3       3a51803cc0d012c3b5dc8b7528cb05f7                      3366   
4       ef0996a1a279c26e7ecbd737be23d235                      2290   
5       e781fdcc107d13d865fc7698711cc572                     88032   
6       3e4fd73f1e86b135b9b121d6abbe9597                     19400   
...                                  ...                

In [19]:
df.to_csv(r"E:\IIMN\MODULE 02\ASSIGNMENT\archive (8)\df_dataframe.csv",index= False)